<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="dustR">
<title>A guide to the dustR package • dustR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A guide to the dustR package">
<meta property="og:description" content="dustR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dustR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/quick_start.html">A guide to the dustR package</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>A guide to the dustR package</h1>
                        <h4 data-toc-skip class="author">Jelmar Quist,
Isobelle Wall</h4>
            
            <h4 data-toc-skip class="date">2024-05-03</h4>
      
      
      <div class="d-none name"><code>quick_start.Rmd</code></div>
    </div>

    
    
<style type="text/css">
  body{
  font-size: 12pt;
}
</style>
<style type="text/css">
h1{color: #13C800 !important}
h2{color: #0072FF !important}
h3{color: #000000 !important}
h4{color: #000000 !important}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this vignette we present <code>dustR</code>, an alternative data
analysis workflow for GeoMx Whole Transcriptome Atlas (WTA) spatial
transcriptomics data. This R pacakge provides a series of functions that
assist in the quality assurance while minimising segment loss and
maximising gene detection rates.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The development version of <code>dustR</code> can be installed from
GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"jelmar-quist/dustR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">dustR</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GeomxTools</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://robustbase.R-forge.R-project.org/" class="external-link">robustbase</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mclust-org.github.io/mclust/" class="external-link">mclust</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://had.co.nz/reshape" class="external-link">reshape</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="preparing-data">Preparing data<a class="anchor" aria-label="anchor" href="#preparing-data"></a>
</h3>
<p>We recommend loading the data similiar to…</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"/Users/k1217281/Documents/GeoMX/EllieAlberts/materials/v2/eset.RData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="no-template-control">No template control<a class="anchor" aria-label="anchor" href="#no-template-control"></a>
</h3>
<p>A No Template Control (NTC) is used to assess contamination of the
segments on the 96-well plate. As a general rule, NTCs with more than
1,000 but less than 10,000 sequencing reads may indicate contamination.
count: values &gt;1,000 could indicate contamination for the segments
associated with this NTC; however, in cases where the NTC count is
between 1,000- 10,000, the segments may be used if the NTC data is
uniformly low (e.g. 0-2 counts for all probes).</p>
<p><code>checkNTC</code> identifies the NTCs in the
<code>NanoStringGeoMxSet</code> object and reports (i) the number of
NTCs containing more than 1,000 sequencing reads, and (ii) the number of
segments that are at risk of contamination.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/checkNTC.html">checkNTC</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Identified 8 out of 8 NTCs with &gt;= 1000 sequencing reads.</span></span></code></pre>
<pre><code><span><span class="co">## 525 out of 525 segments may be at risk, please run checkNTCcounts() before proceeding.</span></span></code></pre>
<p><code>plotNTC</code> can be used to plot the number of sequencing
reads for each NTC.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotNTC.html">plotNTC</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/plotNTC-1.png" width="80%"></p>
<p>Due to the risk of contamination in NTCs with more than 10,000
sequencing reads, we would recommend to discard any segments originating
from that same 96 well plate. For NTCs with more than 1,000, but less
than 10,000 sequencing reads, if the gene count distribution is uniform
(i.e. all genes are present at a similar level within that NTC),
segments from that 96 well plate can be used for further analysis.
<code>checkNTCcounts</code> assess and plot the gene counts distribution
of each NTC.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/checkNTCcounts.html">checkNTCcounts</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/checkNTCcounts-1.png" width="80%"></p>
<p>Based on the findings above, segments likely to be contaminated
should be removed, along with the NTCs. In this vignette, we only remove
the NTCs themselves.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="va">eSet</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"A01"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span>, invert <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="negative-control-probes">Negative control probes<a class="anchor" aria-label="anchor" href="#negative-control-probes"></a>
</h3>
<p>The GeoMx Whole Transcriptome Atlas (WTA) panel includes 139 negative
control probes designed to estimate the background for each segment
which could directly affect gene detection rate and any subsequent
downstream analysis. Note that the negative control probes are designed
to target sequences from the External RNA Controls Consortium (ERCC),
which are not present in the human genome.</p>
<p>With <code>plotNegativeProbes</code>, the slope between overall
abundance of each negative control probe and the number of sequencing
reads is estimated.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotNegativeProbes.html">plotNegativeProbes</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/plotNegativeProbes-1.png" width="80%"></p>
<p>Assuming all negative control probes are present at an equal
abundance, this slope should be more or less similar between the 139
negative control probes. The density plot from
<code>plotNegativeProbes</code> suggests that for the majority of the
negative control probes this indeed the case. However, for 5 negative
control probes, their slope is considerably more steep, resulting in
increased background and reduced gene detection rates. We suggest to
remove these negative control probes using
<code>filterNegativeProbes</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterNegativeProbes.html">filterNegativeProbes</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Removed 5 out of 139 negative control probes due to poor performance.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="processing">Processing<a class="anchor" aria-label="anchor" href="#processing"></a>
</h3>
<p>Following the inspection of the NTCs and the cleaning of negative
control probes, processing of the data can continue.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html" class="external-link">source</a></span><span class="op">(</span><span class="st">"/Users/k1217281/Documents/resources/GeoMx/calculateNegativeProbe.R"</span><span class="op">)</span></span>
<span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GeomxTools/man/shiftCountsOne.html" class="external-link">shiftCountsOne</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span>
<span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu">calculateNegativeProbe</span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p>After processing the data, we inspect four more segment
characteristics to identify poor performing segments, as describes
below.</p>
</div>
<div class="section level3">
<h3 id="nuclei-and-area">Nuclei and area<a class="anchor" aria-label="anchor" href="#nuclei-and-area"></a>
</h3>
<p>The relationship between the number of nuclei and the segment area
provide insight in the performance of the segmentation and the nuclei
count for a given segment. A positive relation between nuclei and area
is expected. Using <code>checkNuclei</code>, segments where this
relationship deviates from the regression model are flagged.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkNuclei.html">checkNuclei</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 0 out of 517 segments were flagged due to abnormal behavior.</span></span></code></pre>
<p>We can further visualise and inspect these findings using
<code>plotNuclei</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotNuclei.html">plotNuclei</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/plotNuclei-1.png" width="80%"></p>
<p>Segments that deviate from the model will be highlighted by a black
outline. In the data presented in this vignette, all segments performed
as expected.</p>
</div>
<div class="section level3">
<h3 id="area-and-sequencing-reads">Area and sequencing reads<a class="anchor" aria-label="anchor" href="#area-and-sequencing-reads"></a>
</h3>
<p>As per the recommendqations, the sequencing libraries should be
prepared with ~100 sequencing reads per µm2 for each segment. By
modelling the relationship between the area and the number of sequencing
reads, segments with less sequencing reads than expected based on the
area can be identified. This could be a result of the tissue quality or
the incorrect pooling of the sequencing libraries.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkRawSequencing.html">checkRawSequencing</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 12 out of 517 segments had less sequencing reads than expected and were flagged.</span></span></code></pre>
<p><code>checkRawSequencing</code> suggests 12 segments had considerably
less sequencing reads than expected based on their area. We can
visualise and inspect this in more detail using
<code>plotRawSequencing</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotRawSequencing.html">plotRawSequencing</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/plotRawSequencing-1.png" width="80%"></p>
<p>This plot suggests that while for for the majority of segments the
number of sequencing reads is as expected, for some, the number of
sequencing reads is less than expected, indiated by a black outline. We
suggest to discard these segments.</p>
</div>
<div class="section level3">
<h3 id="sequencing-saturation">Sequencing saturation<a class="anchor" aria-label="anchor" href="#sequencing-saturation"></a>
</h3>
<p>The sequencing saturation provides information about the quality of
the sequencing and if enoughn was sequenced to capture the full
diversity of the segment. <code>checkSaturation</code> calculates the
sequencing saturation and flags segments with a saturation below 50%
(default).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkSaturation.html">checkSaturation</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Removed 73 out of 517 segments due to low sequencing saturation.</span></span></code></pre>
<p>73 segments were flagged as having reduced sequencing saturation. We
can visualise and inspect the saturation further using
<code>plotSaturation</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSaturation.html">plotSaturation</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/plotSaturation-1.png" width="80%"></p>
<p>While the median saturation is 80%, there are some samples that have
a sequencing saturation below 50%. For further analysis at this time, we
suggest these segments to be discarded. However, in practice, you could
opt to resequence these segments to improve their saturation.</p>
</div>
<div class="section level3">
<h3 id="geometric-mean-of-negative-control-probes">Geometric mean of negative control probes<a class="anchor" aria-label="anchor" href="#geometric-mean-of-negative-control-probes"></a>
</h3>
<p>To calculate the background noise of each segments, the geometric
mean of the negative control probes is used. We can assess the geometric
mean across all segments using <code>plotGeoMeanNegProbes</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotGeoMeanNegProbes.html">plotGeoMeanNegProbes</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/plotGeoMeanNegProbes-1.png" width="80%"></p>
<p>The barplot highlight the large range of the geometric mean across
the segments. Overall, a low geometric mean is considered good, however,
segments with a geometric mean that is too low (default 1) tend to have
a poor gene detection rate. These segments can be flagged using
<code>checkGeoMeanNegProbes</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkGeoMeanNegProbes.html">checkGeoMeanNegProbes</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The average gene detection rate in segments with a NegGeoMean &lt;= 1 is 0.16%</span></span></code></pre>
<pre><code><span><span class="co">## The average gene detection rate in segments with a NegGeoMean &gt; 1 is 18.22%</span></span></code></pre>
<pre><code><span><span class="co">## 17 out of 517were flagged due to a low geometric mean of the negative control probes.</span></span></code></pre>
<p>Not only does this function flag segments with a low geometric mean,
it also calculates the average gene detection rate for these segments
and compares it with the remaining segments. Given that the gene
detection rate in segments with a geometric mean of the negative control
probes is less than 1%, we suggest these to be discarded.</p>
</div>
<div class="section level3">
<h3 id="filter-segments">Filter segments<a class="anchor" aria-label="anchor" href="#filter-segments"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotFilterSegments.html">plotFilterSegments</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_files/figure-html/plotFilterSegments-1.png" width="80%"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterSegments.html">filterSegments</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Removed77 out of 517segments that were flagged.</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="additional-functions">Additional functions<a class="anchor" aria-label="anchor" href="#additional-functions"></a>
</h2>
<div class="section level3">
<h3 id="pseudobulk">Pseudobulk<a class="anchor" aria-label="anchor" href="#pseudobulk"></a>
</h3>
<p><code>createPseudobulk</code> does exactly what the name implies, it
creates pseudobulk data by aggregating the gene counts from all segments
for a given sample. The function also updates the Segment, Nulcei and
Area columns, as well as the reported number of sequencing reads. Sample
definitions are obtained from <code>id.vars</code>, which should
specificy the appropiate column in the pData.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"/Users/k1217281/Documents/GeoMX/EllieAlberts/materials/v2/eset.RData"</span><span class="op">)</span></span>
<span><span class="va">eSet_pseudobulk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createPseudobulk.html">createPseudobulk</a></span><span class="op">(</span><span class="va">eSet</span>, id.vars <span class="op">=</span> <span class="st">"Slide Name"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Samples </span></span>
<span><span class="co">##     525</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">eSet_pseudobulk</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Samples </span></span>
<span><span class="co">##       9</span></span></code></pre>
<p>A second flavour of the function was implemented to generate
pseudobulk for each ROI. I.e. if for a given ROI there are a PanCK and a
CD45 segment, these segments will be collated together. Note that the
function still requires the user to specify the appropiate
<code>id.vars</code> to prevent segments from different samples to be
aggregated.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"/Users/k1217281/Documents/GeoMX/EllieAlberts/materials/v2/eset.RData"</span><span class="op">)</span></span>
<span><span class="va">eSet_pseudobulk</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createPseudobulk.html">createPseudobulk</a></span><span class="op">(</span><span class="va">eSet</span>, id.vars <span class="op">=</span> <span class="st">"Slide Name"</span>, bySlide <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Samples </span></span>
<span><span class="co">##     525</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">eSet_pseudobulk</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Samples </span></span>
<span><span class="co">##     256</span></span></code></pre>
<p>After creating pseudobulk data it is recommended to perform a quality
control.</p>
</div>
<div class="section level3">
<h3 id="spatialge">SpatialGE<a class="anchor" aria-label="anchor" href="#spatialge"></a>
</h3>
<p><code>spatialGE</code> is an R package design for analysing and
visualising spatial transcriptomics data.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">spatialGE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: spatialGE</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"/Users/k1217281/Documents/GeoMX/LEAP_SA_v3/materials/SA_ReadInOnly.RData"</span><span class="op">)</span></span>
<span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="va">eSet</span><span class="op">[</span>,<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">eSet</span><span class="op">$</span><span class="va">Segment</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">eSet</span> <span class="op">&lt;-</span> <span class="va">eSet</span><span class="op">[</span>,<span class="va">eSet</span><span class="op">$</span><span class="va">Segment</span> <span class="op">==</span> <span class="st">"PanCK+"</span><span class="op">]</span></span>
<span></span>
<span><span class="va">eSet_STlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSTlist.html">createSTlist</a></span><span class="op">(</span><span class="va">eSet</span>, id.vars <span class="op">=</span> <span class="st">"LEAP_ID"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #00BB00;">Found list of dataframes.</span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span><span style="color: #BBBB00;">Matching gene expression and coordinate data...</span></span></span>
<span><span class="co"><span style="color: #BBBB00;">## Converting counts to sparse matrices</span></span></span>
<span><span class="co"><span style="color: #BBBB00;">## </span><span style="color: #00BB00; font-weight: bold;">Completed STlist!</span></span></span>
<span><span class="co"><span style="color: #00BB00; font-weight: bold;">## </span></span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jelmar Quist.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
